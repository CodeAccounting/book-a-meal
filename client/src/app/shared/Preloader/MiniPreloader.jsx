import React from 'react';
import Loader from 'react-loaders';
import 'loaders.css/src/animations/ball-beat.scss';
import './Preloader.scss';

const MiniPreLoader = () => (<Loader type="ball-beat" />);

export default MiniPreLoader;
