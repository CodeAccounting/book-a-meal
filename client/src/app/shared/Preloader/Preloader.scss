@import '../../../../public/scss/_variables';
.loader {
    &-hidden {
        display: none;
    }
    &-active {
        display: block;
    }
    .ball-beat>div {
        background-color: $primary-color;
    }
    .ball-scale-multiple {
        // margin: 30vh 0 25vh 46vw;
        height: 11vh;
        width: 11vh;
        &>div {
            left: 0;
        }
    }
}

.modal-preloader {
    display: flex;
    height: 25vh;
    justify-content: center;
    align-items: center;
}