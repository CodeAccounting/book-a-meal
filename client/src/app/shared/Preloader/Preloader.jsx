import React from 'react';
import Loader from 'react-loaders';
import 'loaders.css/src/animations/ball-scale-multiple.scss';
import './Preloader.scss';

const Preloader = () => (<Loader type="ball-scale-multiple" />);

export default Preloader;
