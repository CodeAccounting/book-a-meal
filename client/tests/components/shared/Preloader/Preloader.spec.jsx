import React from 'react';
import Preloader, { MiniPreloader } from '../../../../src/components/shared/Preloader';

describe('Preloader', () => {
  it('renders correctly', () => {
    const shallowWrapper = shallow(<Preloader />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly: MiniPreloader', () => {
    const shallowWrapper = shallow(<MiniPreloader />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
