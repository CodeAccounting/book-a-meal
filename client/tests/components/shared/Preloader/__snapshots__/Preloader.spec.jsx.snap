// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader renders correctly 1`] = `
<div
  className="main-preloader"
>
  <Loader
    active={true}
    type="ball-scale-multiple"
  />
</div>
`;

exports[`Preloader renders correctly: MiniPreloader 1`] = `
<Loader
  active={true}
  type="ball-beat"
/>
`;
